Shoko.Server/Providers/TMDB/TmdbMetadataService.cs

```diff
@@ -2031,6 +2031,50 @@ private void PurgeCompany(int companyId, bool removeImageFiles = true)
 
     #region People
 
+    public async Task<(int Found, int Updated, int Skipped, int Removed)> RepairMissingPersons(bool removeErrors = false)
+    {
+        var missingIds = new HashSet<int>();
+        var (updateCount, skippedCount, removedCount) = (0, 0, 0);
+        
+        var people = _tmdbPeople.GetAll().Select(person => person.TmdbPersonID).ToList();
+        
+        _tmdbEpisodeCast.GetAll().AsParallel()
+            .Where(p => !people.Contains(p.TmdbPersonID))
+            .ForAll(p => missingIds.Add(p.TmdbPersonID));
+        _tmdbEpisodeCrew.GetAll().AsParallel()
+            .Where(p => !people.Contains(p.TmdbPersonID))
+            .ForAll(p => missingIds.Add(p.TmdbPersonID));
+        
+        _tmdbMovieCast.GetAll().AsParallel()
+            .Where(p => !people.Contains(p.TmdbPersonID))
+            .ForAll(p => missingIds.Add(p.TmdbPersonID));
+        _tmdbMovieCrew.GetAll().AsParallel()
+            .Where(p => !people.Contains(p.TmdbPersonID))
+            .ForAll(p => missingIds.Add(p.TmdbPersonID));
+
+        foreach (var personId in missingIds)
+        {
+            try
+            {
+                await UpdatePerson(personId, forceRefresh: true);
+                updateCount++;
+            }
+            catch (NullReferenceException)
+            {
+                _logger.LogDebug("Unable to update TMDB Person ({@PersonId})", personId);
+
+                if (removeErrors && await PurgePerson(personId, removeImageFiles: true, forceRemoval: true))
```

Comment:
It would be better to keep a reference to all the cast/crew roles the person has credits for, and update those, instead of just removing the person. Also don't like that we're just checking for an NPE. Would be better to handle it gracefully instead of throwing, so we should probably update the code below, on line 2051, to expect the remote person to be null-able, and then schedule an update for all needed entities to naturally remove the person from the local database once all references to them are gone.